earthaccess/api.py

```diff
@@ -78,9 +75,7 @@ def search_datasets(
     return query.get_all()
 
 
-def search_data(
-    count: int = -1, **kwargs: Any
-) -> List[earthaccess.results.DataGranule]:
+def search_data(count: int = -1, **kwargs: Any) -> List[DataGranule]:
```

Comment:
In this case, `Any` is the correct type. When typing `kwargs`, the type is the _value_ type. You cannot annotate the _key_ type for `kwargs` because it is a "special" dictionary where you can use _only_ strings (literal keywords) as keys, so you only annotate the _value_ type.

In this case, `Any` is the only reasonable thing we can do because there are no real restrictions on the types of values we can pass for various keywords. Unfortunately, Python does not provide a convenient means for us to fully type a large set of possible keyword arguments.

Reply:
To clarify, in this particular case, we cannot reasonably narrow the value type, but that's not generally true. We might have a case for defining a more specialized function where we expect only values of a certain type(s) for keyword values.